File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/RegexContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser.RegexlikeContext;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.regex.Pattern;

public class RegexContextEvaluator implements ContextEvaluator<RegexlikeContext> {
    private static final Logger logger = LoggerFactory.getLogger(RegexContextEvaluator.class);

    @Override
    public Object evaluate(RegexlikeContext ctx, Visitor visitor) {
        Object value = visitor.visit(ctx.value);

        String regexPattern = ctx.regex.getText().replace("'", "");
        String input = value.toString();

        Pattern pattern = Pattern.compile(regexPattern);
        String s = pattern.matcher(input).replaceAll("");
        logger.debug("Value: '{}' -> '{}'", value, s);
        return s;
    }
}

File: analang-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/RegexContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.analang.ANAParser.RegexlikeContext
import com.rappi.fraud.rules.parser.visitors.Visitor

class RegexContextEvaluator: ContextEvaluator<RegexlikeContext> {
    override fun evaluate(ctx: RegexlikeContext, visitor: Visitor): Any {
        val value = visitor.visit(ctx.value)


        return Regex(ctx.regex.text.replace("'", "")).replace(value.toString(), "")
    }
}